auth/src/ios/credential_ios.mm

```diff
@@ -94,6 +94,48 @@ @implementation PhoneListenerDataObjC
                           : util::StringFromNSString(CredentialFromImpl(impl_).provider);
 }
 
+//
+// PhoneAuthCredential methods
+//
+PhoneAuthCredential::PhoneAuthCredential() {}
+
+PhoneAuthCredential::PhoneAuthCredential(void* impl) {
+  impl_ = impl;
+  error_code_ = kAuthErrorNone;
+}
+
+PhoneAuthCredential::PhoneAuthCredential(const PhoneAuthCredential& rhs) {
+  if (rhs.impl_ != NULL) {
+    impl_ = new FIRPhoneAuthCredentialPointer(PhoneAuthCredentialFromImpl(rhs.impl_));
+  }
+  error_code_ = rhs.error_code_;
+  error_message_ = rhs.error_message_;
+  sms_code_ = rhs.sms_code_;
+}
+
+PhoneAuthCredential& PhoneAuthCredential::operator=(const PhoneAuthCredential& rhs) {
+  if (rhs.impl_ != NULL) {
+    if (impl_ != nullptr) {
+      delete static_cast<FIRPhoneAuthCredentialPointer*>(impl_);
+      impl_ = nullptr;
+    }
+    impl_ = new FIRPhoneAuthCredentialPointer(PhoneAuthCredentialFromImpl(rhs.impl_));
+  }
```

Comment:
What if `rhs.impl_` is `NULL`? I think you should delete `impl_`, if not NULL, and change it to `NULL` as well.

Reply:
I see this is the same implementation in `Credential`. 
https://github.com/firebase/firebase-cpp-sdk/blob/main/auth/src/ios/credential_ios.mm#L72

I think we should fix this in `Credential` and all the child constructor and operator just leverage the implementation in `Credential`.

Reply:
Fixed the `rhs_.impl == nullptr` problem, but I can't reuse the `Credential::operator=` implementation since `Credential` stores a `FIRAuthCredentialPointer` in the `void* impl`instead of the `FIRPhoneAuthCredentialPointer` here.

Reply:
I can reuse the implementation of `Credential::operator=` on the Android-side, though.

Reply:
I see. Thanks for the update.

Could you also fix `Credential::operator=()` and make sure it deletes `impl_` as well very similar to what you are doing here. 
I think the current `Credential::operator=()` is not correct.

Reply:
Done.